core/src/main/scala/com/microsoft/ml/spark/codegen/DotnetWrappable.scala

```diff
@@ -0,0 +1,522 @@
+  }
+
+  //noinspection ScalaStyle
+  protected def dotnetParamSetter(p: Param[_]): String = {
+    val capName = p.name match {
+      case "xgboostDartMode" => "XGBoostDartMode"
+      case "parallelism" => dotnetClassName match {
+        case "VowpalWabbitContextualBandit" => "ParallelismForParamListFit"
+        case _ => p.name.capitalize
+      }
+      case _ => p.name.capitalize
+    }
+    val docString =
+      s"""|/// <summary>
+          |/// Sets ${p.name} value for <see cref=\"${p.name}\"/>
+          |/// </summary>
+          |/// <param name=\"${p.name}\">
+          |/// ${p.doc}
+          |/// </param>
+          |/// <returns> New $dotnetClassName object </returns>""".stripMargin
+    p match {
+      case sp: ServiceParam[_] =>
+        s"""|$docString
+            |public $dotnetClassName Set$capName(${getServiceParamInfo(sp).dotnetType} value) =>
+            |    $dotnetClassWrapperName(Reference.Invoke(\"set$capName\", (object)value));
+            |
+            |public $dotnetClassName Set${capName}Col(string value) =>
+            |    $dotnetClassWrapperName(Reference.Invoke(\"set${capName}Col\", value));
+            |""".stripMargin
+      case _: DataTypeParam =>
+        s"""|$docString
+            |public $dotnetClassName Set$capName(${getParamInfo(p).dotnetType} value) =>
+            |    $dotnetClassWrapperName(Reference.Invoke(\"set$capName\",
+            |    DataType.FromJson(Reference.Jvm, value.Json)));
+            |""".stripMargin
+      case _: StringStringMapParam | _: StringIntMapParam =>
+        s"""|$docString
+            |public $dotnetClassName Set$capName(${getParamInfo(p).dotnetType} value)
+            |{
+            |    var hashMap = new HashMap(SparkEnvironment.JvmBridge);
+            |    foreach (var item in value)
+            |    {
+            |        hashMap.Put(item.Key, item.Value);
+            |    }
+            |    return $dotnetClassWrapperName(Reference.Invoke(\"set$capName\", (object)hashMap));
+            |}
+            |""".stripMargin
+      case _: EstimatorParam | _: ModelParam =>
+        s"""|$docString
+            |public $dotnetClassName Set${capName}<M>(${getParamInfo(p).dotnetType} value) where M : ScalaModel<M> =>
+            |    $dotnetClassWrapperName(Reference.Invoke(\"set$capName\", (object)value));
+            |""".stripMargin
+      case _: ArrayParamMapParam | _: TransformerArrayParam | _: EstimatorArrayParam | _: UntypedArrayParam =>
+        s"""|$docString
+            |public $dotnetClassName Set$capName(${getParamInfo(p).dotnetType} value)
+            |{
+            |    var arrayList = new ArrayList(SparkEnvironment.JvmBridge);
+            |    foreach (var v in value)
+            |    {
+            |        arrayList.Add(v);
+            |    }
+            |    return $dotnetClassWrapperName(Reference.Invoke(\"set$capName\", (object)arrayList));
+            |}
+            |""".stripMargin
+      case _: ArrayMapParam =>
```

Comment:
Likewise here: we might want to put this into WrappableParam so that the code generator is slim and the implementations are distributed

Reply:
Ah do you mean putting the whole function into WrappableParam?